### PR TITLE
feat(Repository/Query_Filters.php) support after order clauses

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/133303

This PR is the base for a modification I've done to TEC to allow for our custom `orderby` clauses to:

* not apply in exclusive mode
* apply before or after WordPress clauses

Right now this code:

```php
$events = tribe_events()->orderby( ['event_date' => 'ASC', 'menu_order' => 'ASC' ] )->get_ids();
```

Would order events **only** by `event_date`; the `menu_order` clause would be dropped entirely.  
With the modified code in the PR the `ORDER` clauses would be applied like this:

* order by `event_date ASC` then...
* order by `menu_order ASC`

To move the `event_date` orderby clause after the WordPress-managed `menu_order` one we can prefix order keys we control (`event_date`, `event_date_utc`, `venue`, `organizer`) with the `Tribe__Repository__Query_Filters::AFTER` constant:

```php
$event_date = Tribe__Repository__Query_Filters::AFTER . 'event_date';
$events = tribe_events()->orderby( [ $event_date => 'ASC', 'menu_order' => 'ASC' ] )->get_ids();
```

To have the `ORDER` clauses apply like this:

* order by `menu_order ASC` then...
* order by `event_date ASC` 

What is **not** in this PR: while this improves on what we have, this does not allow setting order (ASC, DESC) per order by criteria. This could be part of another PR if we need to.

### CODE REVIEW

**Comment by lucatume** [comment]
@bordoni  and @sc0ttkclark  I'm requesting you both as the Repository is a quite sensible contact area between teams and I would like both to give me the green light on this.

**Comment by lucatume** [comment]
After our discussion on Friday (10/18) I've modified this code so that the `Query_Filters::orderby` method now supports more complex `$orderby` arguments.

The best example of possible usages, being this a low level API, can be found in the tests for this PR code.

File: src/Tribe/Repository/Query_Filters.php [hunk 1]
```diff
@@ -693,18 +700,42 @@ public function join( $join_clause, $id = null, $override = false ) {
...
+		// Use the `$id` parameter to allow later method calls to replace values set in previous calls.
 		if ( $id ) {
-			if ( $override || ! isset( $this->query_vars['orderby'][ $id ] ) ) {
-				$this->query_vars['orderby'][ $id ] = $orderby;
+			if ( $override || ! isset( $this->query_vars[ $orderby_key ][ $id ] ) ) {
+				$this->query_vars[ $orderby_key ][ $id ] = $entries;
 			}
 		} else {
-			$this->query_vars['orderby'][] = $orderby;
+			$this->query_vars[ $orderby_key ][$id] = array_merge( $this->query_vars[ $orderby_key ][$id], $entries );
```
Comment (sc0ttkclark) [inline src/Tribe/Repository/Query_Filters.php]:
```suggestion
			$this->query_vars[ $orderby_key ][ $id ] = array_merge( $this->query_vars[ $orderby_key ][ $id ], $entries );
```

**Comment by lucatume** [comment]
@bordoni this still needs a ticket, the one you mentioned. 
I'm merging when the tests pass and left you a note about the ticket.

**Comment by lucatume** [comment]
> I like the code here, but we do need a changelog as it affects current codebase, as well a ticket for QA to test the old Widgets and Views for ordering.

I've created [this ticket to track the effort](https://central.tri.be/issues/136256).

**Pull request merged**